src/main/java/rx/internal/util/ObjectPool.java

```diff
@@ -18,20 +18,15 @@
 package rx.internal.util;
 
 import java.util.Queue;
-import java.util.concurrent.ConcurrentLinkedQueue;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
 
-import rx.Scheduler;
-import rx.functions.Action0;
-import rx.internal.util.unsafe.MpmcArrayQueue;
-import rx.internal.util.unsafe.UnsafeAccess;
-import rx.schedulers.Schedulers;
+import rx.internal.util.unsafe.*;
 
 public abstract class ObjectPool<T> {
     private Queue<T> pool;
     private final int maxSize;
 
-    private Scheduler.Worker schedulerWorker;
+    private final ScheduledExecutorService schedulerWorker;
```

Comment:
We should combine all periodic maintenance-related threads into a single service: ObjectPool, IO scheduler's cleanup pool, JDK 6 computation scheduler's purge.